cmd/promtool/resource.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/google/pprof/profile"
+)
+
+type resourceSaverConfig struct {
+	server         string
+	tarballName    string
+	pathToFileName map[string]string
+	postProcess    func(b []byte) (*bytes.Buffer, error)
+}
+
+type resourceSaver struct {
+	archiver
+	httpClient        httpClient
+	fileNameToRequest map[string]*http.Request
+	postProcess       func(b []byte) (*bytes.Buffer, error)
+}
+
+func newResourceSaver(cfg resourceSaverConfig) *resourceSaver {
+	client, err := newHTTPClient(httpClientConfig{serverURL: cfg.server})
+	if err != nil {
+		panic(err)
+	}
+	a, err := newArchiver(archiverConfig{archiveName: cfg.tarballName})
+	if err != nil {
+		panic(err)
+	}
+	m := make(map[string]*http.Request)
+	for path, filename := range cfg.pathToFileName {
+		req, err := http.NewRequest(http.MethodGet, client.urlJoin(path), nil)
+		if err != nil {
+			panic(err)
+		}
+		m[filename] = req
+	}
+	return &resourceSaver{
+		a,
+		client,
+		m,
+		cfg.postProcess,
```

Comment:
same here I think we need more descriptive var names , reqs/requests , archiver are not so long.